src/audio/types.h

```diff
@@ -244,6 +244,53 @@ class Bitrate {
 
 QDebug operator<<(QDebug dbg, Bitrate arg);
 
+// The BarLength is measured in beats and provides information
+// about the number expected within a single bar or phrase.
+class BarLength {
+  public:
+    using value_t = std::uint32_t;
+
+    static BarLength valueFromUInt(unsigned int value) {
+        VERIFY_OR_DEBUG_ASSERT(value >= std::numeric_limits<value_t>::min() &&
+                value <= std::numeric_limits<value_t>::max()) {
+            return BarLength(0);
+        }
+        return BarLength(static_cast<value_t>(value));
+    }
+
+  private:
+    // The default value is invalid and indicates a missing or unknown value.
```

Comment:
To be honest, I think 4/4 is a reasonable assumption. Otherwise you'd require people to explicitly edit the beatgrid for every single track. I'm more worried about the default downbeat location, because I'm not sure how many tracks start with the first beat in a bar (rather than the 4th beat as intro).

But i think most of this can be improved later on, not sure if this needs to be handled in this PR.

Reply:
I agree that 4/4 is the default and that we should not require people to set that. 

Can we later distinguish if the track has the guessed 4/4 setting or confirmed and start adjusted setting? This is IMHO mandatory to improve this later on .

Reply:
We can - The double-beat mechanism is only used if a marker (or a bar length) was explicitely set.